From 9da42f8d7c3c5038c5a09b1730a70efa645bdce5 Mon Sep 17 00:00:00 2001
From: "Adam W. Willis" <return.of.octobot@gmail.com>
Date: Sat, 24 May 2025 21:39:37 -0400
Subject: [PATCH 1/4] KernelSU: Support CMD_HOOK_MODE for KSU-Next

Maintain feature parity subsequent to KernelSU-Next/KernelSU-Next@ce37e17

Signed-off-by: Adam W. Willis <return.of.octobot@gmail.com>
---
 kernel/core_hook.c | 8 ++++++++
 kernel/ksu.h       | 1 +
 2 files changed, 9 insertions(+)

diff --git a/kernel/core_hook.c b/kernel/core_hook.c
index 4c2e6d35..5ae5b6b2 100644
--- a/kernel/core_hook.c
+++ b/kernel/core_hook.c
@@ -382,6 +382,14 @@ int ksu_handle_prctl(int option, unsigned long arg2, unsigned long arg3,
 		return 0;
 	}
 
+	if (arg2 == CMD_HOOK_MODE) {
+		const char *mode = "Manual";
+		if (copy_to_user((void __user *)arg3, mode, strlen(mode) + 1)) {
+			pr_info("hook: copy_to_user() failed\n");
+		}
+		return 0;
+	}
+
 	if (arg2 == CMD_REPORT_EVENT) {
 		if (!from_root) {
 			return 0;
diff --git a/kernel/ksu.h b/kernel/ksu.h
index e5050024..af81a43d 100644
--- a/kernel/ksu.h
+++ b/kernel/ksu.h
@@ -23,6 +23,7 @@
 #define CMD_UID_SHOULD_UMOUNT 13
 #define CMD_IS_SU_ENABLED 14
 #define CMD_ENABLE_SU 15
+#define CMD_HOOK_MODE 16
 
 #define EVENT_POST_FS_DATA 1
 #define EVENT_BOOT_COMPLETED 2
-- 
2.43.0

